app/models/deployment.rb

```diff
@@ -1,17 +1,32 @@
+# Instances of this class represent state in Github's API.
 # Deployment represents a created DeploymentRequest.
 class Deployment
   include Virtus.value_object
 
   values do
     # The external id of the deployment.
     attribute :id, Integer
-    # The name of the repository the deployment was for.
-    attribute :repository, String
+    # The external url of the deployment, needed to lookup statuses.
+    attribute :url, String
     # The ref that was requested to be deployed.
     attribute :ref, String
     # The commit sha that the ref was resolved to (what actually got deployed).
     attribute :sha, String
     # The environment that was deployed to.
     attribute :environment, String
+    ## The name of the repository the deployment was for.
+    attribute :repository, String
+  end
+
+  # memoize the parts of this deployment's url.
+  def url_parts
+    # ["https:", "", "api.github.com", "repos", "octocat", "example", "deployments", "1"]
+    @parts ||= url.split("/").to_a
+  end
+
+  # return the Github Org for deployment.
+  def organization
```

Comment:
This seems a little error prone. Let's follow the same pattern we do in the `Repository` model:

https://github.com/remind101/slashdeploy/blob/2dd90f45b07affafe2e91f34b6f715201e6a2f2e/app/models/repository.rb#L49-L52

And match on `self.repository`.